### ISSUE TITLE
Can’t change x-axis on web interface plots 

### ISSUE THREAD

**Issue by dlebauer** [PecanProject/pecan#3328]
### Bug Description

Web interface graphs always have time on the x-axis - changing the X-axis variable seems to have no effect

### To Reproduce

1. Go to pecan web interface https://pecan-dev.ncsa.cloud/pecan/08-finished.php?workflowid=99000000001&hostname=docker&loglines=20
2. Conduct a run or go to an existing successful run from https://pecan-dev.ncsa.cloud/pecan/history.php
3. Click on graphs —> x-axis —> anything but ’time’
4. Click ‘plot’
5. Notice that x axis is always’time’

## Expected behavior

Changing variable on x axis should change plot 

## Screenshots

![image](https://github.com/PecanProject/pecan/assets/464871/c9090e7c-d540-41ac-91e3-bc607536576e)

## Machine (please complete the following information):

 - Server pecan-dev.ncsa.cloud
 - OS: Linux
 - Version PEcAn v1.8.0

## Additional context

not sure if this is a necessary feature - usually we want ‘time’ on x axis, this is mostly for diagnostics, and data is available for download so that users can do custom analysis.

**Comment by harshagr70** [comment]
@dlebauer i would like to work on this issue , can you please help me out with the links they seems to be broken !!

**Comment by dlebauer** [comment]
Sorry @harshagr70 - ~I think just the base urls need to be updated to http://pecan.ncsa.illinois.edu, i'll update the links in the OP~ I'll have to defer to Rob, as now I am a) not sure that is the correct server and b) getting an error after submitting a job to recreate the example: 

> Warning: mkdir(): No such file or directory in /var/www/html/pecan/04-runpecan.php on line 217
Can't create output folder [/data/workflows/PEcAn_99000000001]

Also if the URL is supposed to be https://pecan-dev.ncsa.cloud/pecan, then we should update the link at 'documentation --> demo' on the pecanproject.org website

**Comment by harshagr70** [comment]
@dlebauer i am trying to set this and run the server locally , i will post any blockers here !! also i need to recheck for the correct link !!

**Comment by AritraDey-Dev** [comment]
Hi @dlebauer,
I recently noticed that the API was always using `time as the x-axis` variable regardless of the input from the web UI. I’ve fixed this on my side so it now correctly respects the x_var parameter. I can raise a quick PR with the fix. I’m also attaching some screenshots showing how the graphs look after the fix. Please let me know if you’d like me to proceed!

![Image](https://github.com/user-attachments/assets/b6627627-b81f-4e16-b2cd-15fa4f47db0b)

![Image](https://github.com/user-attachments/assets/0d35cb5d-6db1-4f4e-b055-308eaea3fefd)

**Comment by harshagr70** [comment]
You are right !! @AritraDey-Dev  , the x_var expects to be passed as in query parameter , otherwise it is framed to fall back to the default value of time , which throws out this error .

<img width="911" alt="Image" src="https://github.com/user-attachments/assets/fc8a048b-880f-4e1d-a0de-347b5aff81c4" />
I was ready for the fix but anyways , if you have the fix ready you can go ahead and raise a PR .

**Comment by AritraDey-Dev** [comment]
Hi @harshagr70, if you have a moment, could you kindly take a look and cross-check the approach here #3528  I used to extract the `xvar` value from the `req` object?

**Comment by harshagr70** [comment]
@AritraDey-Dev , the fix here looks good ! , was going to use the same logic to standalone extract the value from the req rather then expecting it to be passed as an argument in the function .

**Comment by AritraDey-Dev** [comment]
closing this.Resolved by #3528.
